src/main/java/org/dzianisbova/infrastructure/load/SequentialRequestExecutor.java

```diff
@@ -0,0 +1,74 @@
+package org.dzianisbova.infrastructure.load;
+
+import org.dzianisbova.domain.api.Request;
+import org.dzianisbova.domain.load.RequestExecutor;
+import org.dzianisbova.domain.logging.Logger;
+import org.dzianisbova.domain.metrics.StatisticsService;
+import org.dzianisbova.domain.response.Response;
+
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+
+public class SequentialRequestExecutor implements RequestExecutor {
+    private final Logger logger;
+    private final HttpClient httpClient;
+    private final StatisticsService statistic;
+
+    public SequentialRequestExecutor(Logger logger, HttpClient httpClient, StatisticsService statistic) {
+        this.logger = logger;
+        this.httpClient = httpClient;
+        this.statistic = statistic;
+    }
+
+    @Override
+    public Response execute(Request request) {
+        String fullUrl = buildFullUrl(request);
+        HttpRequest httpRequest = buildHttpRequest(request, fullUrl);
+        long startTime = System.currentTimeMillis();
+        try {
+            HttpResponse<String> httpResponse = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+            long duration = System.currentTimeMillis() - startTime;
+            Response response = new Response(httpResponse.statusCode(), httpResponse.body(), duration);
+            logger.info(request, response);
+            statistic.recordSuccess(duration);
+
+            return response;
+        } catch (Exception e) {
+            long duration = System.currentTimeMillis() - startTime;
+            logger.error(request, duration, e);
+            statistic.recordError(duration);
+
+            return new Response(-1, e.getMessage(), duration);
+        }
+    }
+
+    private String buildFullUrl(Request request) {
```

Comment:
Skoro mamy wiele implementacji RequestExecutor i spodziewamy się że raczej to jest HTTP, to może ta metoda powinna trafić do klasy abstakcyjnej? Podobnie z buildHttpRequest